src/app.jsx

```diff
@@ -49,6 +72,11 @@ class App extends React.Component {
     }
     this.setState(this.state);
   }
+  syncData() {
```

Comment:
syncing var happens in the `componentwillreceiveprops` no need for explicit assignment.